### PR TITLE
elastic-opentelemetry-instrumentation-openai: add integrations tests and test on baseline

### PR DESCRIPTION
**Pull request by xrmx**
## What does this pull request do?

This adds a bunch of integration tests hitting openai. This requires
setting a variable OPENAI_API_KEY in github secrets. Tests themselves
are a copy of the recorded ones asserting values from the response data.

Integration tests are not run by default and can be called with pytest --integration-tests

Then adds unit testing against 1.2.0 baseline. 1.26.0 is required for having usage data with streams but with some
small changes we can keep the baseline as old as 1.2.0.

### CODE REVIEW

**Review by trentm — APPROVED** [review]

File: instrumentation/elastic-opentelemetry-instrumentation-openai/tests/test_chat_completions.py [hunk 1]
```diff
@@ -872,6 +874,146 @@ def test_tools_with_capture_content_log_events(
...
+    metrics_reader,
+    request,
+):
+    provider = request.getfixturevalue(provider_str)
+    client = provider.get_client()
+
+    # Redo the instrumentation dance to be affected by the environment variable
+    OpenAIInstrumentor().uninstrument()
+    with mock.patch.dict(
+        "os.environ", {"ELASTIC_OTEL_GENAI_CAPTURE_CONTENT": "true", "ELASTIC_OTEL_GENAI_EVENTS": "log"}
```
Comment (codefromthecrypt) [inline instrumentation/elastic-opentelemetry-instrumentation-openai/tests/test_chat_completions.py:903]: I know this would be in a different PR, but I think we should rename this variable to be the same as upstream, right?

ELASTIC_OTEL_GENAI_CAPTURE_CONTENT -> OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT

Comment (codefromthecrypt) [reply instrumentation/elastic-opentelemetry-instrumentation-openai/tests/test_chat_completions.py:903]: also log are default in 1.28.0, so we should switch precedence

Comment (xrmx) [reply instrumentation/elastic-opentelemetry-instrumentation-openai/tests/test_chat_completions.py:903]: Yep, Will take care of these on a later PR.
